### ISSUE TITLE
Types are not used

### ISSUE THREAD

**Issue by leoseccia** [docusign/docusign-csharp-client#165]
Throughout the application, all variables are of type `string` . This is a real shame - I suggest changing the variable types to reflect the data they hold. 

As it is at the moment, the api is unnecessarily confusing to use because of it.

**Comment by Socrathustra** [comment]
Seconded. I find it frustrating to have to know the magic string values for things like status. At the very least, significant values ought to be enums, constants, or static readonly variables. Adopting the "no magic strings" policy has dramatically increased the quality of my own code and would do wonders for this library.

**Comment by LarryKlugerDS** [comment]
Yes, we agree! This enhancement is on our roadmap but is not yet scheduled. It is filed as internal ticket EC-1104.

Please ask your DocuSign business or technical contact to add your organization's name to the ticket. Thank you for raising the issue.

**Comment by adamhathcock** [comment]
I'd like this for at least scalar types like DateTime or DateTimeOffset
